### ISSUE TITLE
Get rid of stupid memcpy calls

### ISSUE THREAD

**Issue by atoms118** [atoms118/libpolynet-c#14]
There's a whole bunch of calls to `memcpy` where I could easily just use arrays like normal (as in `somearray[index] = somevalue`, rather than `memcpy(somearray + index, somevalue, sizeof(typeof somevalue))`, because the latter is obviously stupid)

This is a good first issue, both for people who haven't worked on this project, or even for people who don't have much experience with C (although an understanding of pointers is required)

**Comment by atoms118** [comment]
I'm putting this on hold until we get #23 out of the way

**Comment by bvs23bkv33** [comment]
i can do this
1) shoud i fork and then ask for pull request?
2) how to test this?

**Comment by atoms118** [comment]
Help would be great, yeah!

@bvs23bkv33 I'll add you to the contributors so you can just create a new branch and PR it

In terms of testing, currently the only testing we have is checking that it compiles. I *think* compile instructions are in the readme, but in case I forgot them, all you need to do is run `make`

In terms of this specific issue, it's sort of dependent on #23, but if you give me awhile I can either get #23 done or give you a list of functions that aren't affected by #23. It should only take me ~1 day (of course, if you want to just take over #23 that works too)

Thanks for taking interest

**Comment by atoms118** [comment]
Ok so something else I forgot about is the `pn_remove()` function, so I've created another issue #29, since this is also going to be heavily dependent on that getting done as well as #23
